torchvision/datasets/imagenet.py

```diff
@@ -46,14 +44,23 @@ class ImageNet(ImageFolder):
         imgs (list): List of (image path, class_index) tuples
         targets (list): The class_index value for each image in the dataset
     """
+    def __init__(self, root, split='train', download=None, **kwargs):
+        if download is None:
+            download = False
+        else:
+            msg = ("The use of the download flag is deprecated, since the public "
```

Comment:
Given that the dataset is not available anymore, I think we can just raise an error if `download=True`.
Plus, I think that we could try being a bit smarter in the extraction code, and instead of requiring the user to manually call the `parse_devkit_archive` etc, just check the following:
- if there is an extracted folder, do nothing
- if there is a file named `parse_*_archive`, extract it to the `root` folder and return
- else raise an error saying to download the files in the website.

Reply:
> Given that the dataset is not available anymore, I think we can just raise an error if `download=True`.

Fine by me.

---

> Plus, I think that we could try being a bit smarter in the extraction code, and instead of requiring the user to manually call the `parse_devkit_archive` etc, just check the following:

So do this check at creation of the dataset?

---

> if there is an extracted folder, do nothing

How would you handle the devkit since there is no folder after parsing? Simply check for the meta file?

---

> if there is a file named `parse_*_archive`, extract it to the root folder and return

What would be the content of this file and how would it be created? Or did you mean the archives?

Reply:
> So do this check at creation of the dataset?

Yes.

> How would you handle the devkit since there is no folder after parsing? Simply check for the meta file?

Just check for the extracted file if it's a single file

> What would be the content of this file and how would it be created? Or did you mean the archives?

Those are the original zip files that have been downloaded by the user.

My thinking is the following:
- user tries the dataset (without having it).
- code raises an error saying to download the zip files from the website and put it in `root`
- code checks for the presence of those files and extracts it

this way, we abstract away the need for the manual extraction.

Let me know what you think

Reply:
This is basically what I've implemented for the `download` flag BC.

Reply:
Cool, that sounds good to me, thanks!